app/Jobs/GetDataFromKobo.php

```diff
@@ -136,76 +132,76 @@ public function handle()
-                                    }
-                                }
-                                if($cultivo['problema'] == 'plagas' || $cultivo['enfermedades'] == 'ambas') {
-                                    $cultivo_module = 'enfermedades';
-                                    $dataMap_cultivo_module = DataMap::findorfail($cultivo_module);
-                                    $new_module = DataMapController::newRecord($dataMap_cultivo_module, $cultivo);  
-
-                                }
-
-                            } else {
-                                $dataMap_cultivo_module = DataMap::findorfail($cultivo_module);
-                                $new_module = DataMapController::newRecord($dataMap_cultivo_module, $cultivo);  
-                                
-                            }  
+                // Iterate through the other parcela-level modules.
+                foreach ($newSubmission['modulos'] as $parcelaModule) {
+                    if ($parcelaModule === 'C') {
+                        $this->processCultivoData($newSubmission);
+                    } else {
+                        $dataMap = DataMap::findOrFail($parcelaModule);
+                        $data = $newSubmission['modulo_loop'];
+                        $data['parcela_id'] = $newSubmission['parcela_id'];
+                        $data['_id'] = $newSubmission['_id'];
+
+                        DataMapController::newRecord($dataMap, $data);
+                    }
+                } // end foreach plot-level module
+            } // end if $submission does not exist
+        } // end foreach record
+    } // end handle method
+
+    public function processCultivoData($newSubmission)
+    {
+        foreach ($newSubmission['modulo_loop']['cultivo_loop'] as $cultivo) {
+            $dataMap = DataMap::findorfail('C');
+
+            $cultivo['modulo_cultivo_loop'] = $cultivo['modulo_cultivo_loop'][0] + $cultivo['modulo_cultivo_loop'][1];
+            $cultivo['parcela_id'] =  $newSubmission['parcela_id'];
+            $cultivo['_id'] =  $newSubmission['_id'];
+
+            // get the cultivo_id from the creation of the cultivo
+            $new_cultivo = DataMapController::newRecord($dataMap, $cultivo);
+            $cultivo['cultivo_id'] =  $new_cultivo->id;
+
+            $cultivo_modules = $cultivo['modulos_cultivo'] . ' '. $cultivo['modulo_cultivo_loop']['extra_modulo_cultivo'];
+            $cultivo_modules = explode(' ', $cultivo_modules);
+            $cultivo = $cultivo + $cultivo['modulo_cultivo_loop'];
+            unset($cultivo['modulo_cultivo_loop']);
+
+            foreach ($cultivo_modules as $cultivo_module) {
+                if ($cultivo_module == 3) {
+                    if ($cultivo['problema'] == 'plagas' || $cultivo['problema'] == 'ambas') {
+                        foreach ($cultivo['plaga_repeat'] as $plaga) {
+                            $dataMap_cultivo_module = DataMap::findorfail('plagas');
```

Comment:
Given we are doing a hard-coded check for the "plagas" data map, I think it's easier to read without first setting it as a variable.